debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/VmCheckCommand.java

```diff
@@ -149,11 +149,13 @@ private void checkJ9VMThreadSanity(J9JavaVMPointer javaVM, PrintStream out) thro
 
 			verifyJ9VMThread(out, thread, javaVM);
 
-			if (thread.publicFlags().allBitsIn(J9Consts.J9_PUBLIC_FLAGS_VM_ACCESS)) {
-				numberOfThreadsWithVMAccess++;
+			if (!J9BuildFlags.J9VM_INTERP_ATOMIC_FREE_JNI || thread.inNative().eq(0)) {
```

Comment:
That's a particularly awful way of doing this. At this point, we could just omit the atomic-free ifdef as inNative is always available and will be 0 if the build flag is false.

Reply:
Or write it as nested ifs...

Reply:
I understand it can't be done in the same sleazy way as the C code, so I guess I can accept this.

Reply:
Ok, I'll remove the `J9VM_INTERP_ATOMIC_FREE_JNI` part of the condition.

Reply:
Done.